File: 프로그래머스/lv3/17676. ［1차］ 추석 트래픽/［1차］ 추석 트래픽.js
function solution(lines) {
  const timeLogs = lines.map((line) => {
    const [date, time, duration] = line.split(" ");
    const endDate = new Date(`${date}T${time}`);
    const durationMs = parseFloat(duration.replace("s", "")) * 1000;
    const startDate = new Date(endDate.getTime() - durationMs + 1); // 시작 시간 계산
    return [startDate.getTime(), endDate.getTime()];
  });

  let maxCount = 0;
  for (let i = 0; i < timeLogs.length; i++) {
    const curEndTime = timeLogs[i][1];
    let curCount = 0;
    for (let j = 0; j < timeLogs.length; j++) {
      const [start, end] = timeLogs[j];
      if (start < curEndTime + 1000 && end >= curEndTime) {
        curCount++;
      }
    }
    maxCount = Math.max(maxCount, curCount);
  }
  return maxCount;
}

File: 프로그래머스/lv3/17676. ［1차］ 추석 트래픽/［1차］ 추석 트래픽.py
from datetime import datetime, timedelta

def solution(lines):
    timeLogs = []
    for line in lines:
        date, time, duration = line.split(" ")
        endDate = datetime.strptime(date + "T" + time, "%Y-%m-%dT%H:%M:%S.%f")
        durationMs = float(duration[:-1]) * 1000
        startDate = endDate - timedelta(milliseconds=durationMs) + timedelta(milliseconds=1)
        timeLogs.append([startDate.timestamp() * 1000, endDate.timestamp() * 1000])
    
    maxCount = 0
    for i in range(len(timeLogs)):
        curEndTime = timeLogs[i][1]
        curCount = 0
        for j in range(len(timeLogs)):
            start, end = timeLogs[j]
            if start < curEndTime + 1000 and end >= curEndTime:
                curCount += 1
        maxCount = max(maxCount, curCount)
    
    return maxCount
